### PR TITLE
apis/engines: Revise the clear() buffer behavior.

### PR DESCRIPTION
**Pull request by hermet**
ThorVG has offered an option to clear the buffer since version 1.0. This is essential when users utilize the canvas target buffer with the main render target. They share the buffer and need to draw contents onto the existing contents.

API:
Result Canvas::clear(bool free = true)
-> Result Canvas::clear(bool free = true, bool buffer = true)

Issue: https://github.com/thorvg/thorvg/issues/1779

### CODE REVIEW

File: src/renderer/gl_engine/tvgGlRenderTask.cpp [hunk 1]
```diff
@@ -111,11 +111,10 @@ void GlComposeTask::run()
     GL_CHECK(glBindFramebuffer(GL_FRAMEBUFFER, getSelfFbo()));
 
     // clear this fbo
-    GLenum color_buffer = GL_COLOR_ATTACHMENT0;
-    const float transparent[] = {0.f, 0.f, 0.f, 0.f};
-
-    GL_CHECK(glDrawBuffers(1, &color_buffer));
```
Comment (hermet) [inline src/renderer/gl_engine/tvgGlRenderTask.cpp:117]: @RuiwenTang  removed this color buffer.

File: src/renderer/wg_engine/tvgWgRenderer.cpp [hunk 2]
```diff
@@ -273,9 +274,13 @@ bool WgRenderer::sync() {
             WGPURenderPassColorAttachment colorAttachment{};
             colorAttachment.view = backBufferView;
             colorAttachment.resolveTarget = nullptr;
-            colorAttachment.loadOp = WGPULoadOp_Clear;
+            if (mClearBuffer) {
```
Comment (hermet) [inline src/renderer/wg_engine/tvgWgRenderer.cpp:277]: @SergeyLebedkin Please confirm

File: src/renderer/gl_engine/tvgGlRenderTask.cpp (hunk 1 above)
Comment (RuiwenTang) [reply src/renderer/gl_engine/tvgGlRenderTask.cpp:117]: If examples/Animation can be rendered correctly, I think it can be removed.

Comment (hermet) [reply src/renderer/gl_engine/tvgGlRenderTask.cpp:117]: Sure, it works.

**Review by RuiwenTang — APPROVED** [review]

**Comment by hermet** [comment]
@gaelyvan We gonna apply this change for skipping buffer clearing. Please let us know if this doesn't work for you.

File: src/renderer/wg_engine/tvgWgRenderer.cpp (hunk 2 above)
Comment (SergeyLebedkin) [reply src/renderer/wg_engine/tvgWgRenderer.cpp:277]: colorAttachment.clearValue = { 0.0, 0.0, 0.0, 1.0 };
move floats to doubles
alpha channel should be 1.0, i guess

Comment (hermet) [reply src/renderer/wg_engine/tvgWgRenderer.cpp:277]: I intentionally changed it. the cleared buffer should become total transparent, should not have any alpha values. So the previous is inccorect.

Comment (SergeyLebedkin) [reply src/renderer/wg_engine/tvgWgRenderer.cpp:277]: OK, And by the spec color channels values have a double type

**Pull request merged**